.github/actions/unit-tests/action.yml

```diff
@@ -0,0 +1,59 @@
+name: Run unit tests
+
+inputs:
+  python_version:
+    required: true
+  GH_USER_ADMIN:
+    required: true
+  GH_TOKEN_ADMIN:
+    required: true
+  
+runs:
+  using: composite
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: debug input
+      shell: bash
+      run: echo '${{ inputs.python_version }}'
+
+    - name: Setup python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python_version }}
+
+    - name: Setup addon
+      shell: bash
+      run: |
+        if [ -f "poetry.lock" ]
+          then
+            mkdir -p package/lib || true
+            pip install poetry==1.5.1 poetry-plugin-export==1.4.0
+            poetry export --without-hashes -o package/lib/requirements.txt
+            poetry export --without-hashes --dev -o requirements_dev.txt
+          fi
+        if [ ! -f requirements_dev.txt ]; then echo no requirements;exit 0 ;fi
+        git config --global url."https://${{ inputs.GH_TOKEN_ADMIN }}@github.com".insteadOf https://github.com
```

Comment:
This command is needed to properly install dependencies from internal repos. Discovered while working on [https://splunk.atlassian.net/browse/ADDON-66730?atlOrigin=eyJpIjoiMDdlM2ZhMTE4NDMwNDViZDkyOGNhZWRmYzdiZjcwYzIiLCJwIjoiaiJ9](https://splunk.atlassian.net/browse/ADDON-66730?atlOrigin=eyJpIjoiMDdlM2ZhMTE4NDMwNDViZDkyOGNhZWRmYzdiZjcwYzIiLCJwIjoiaiJ9)

Reply:
https://github.com/srv-rr-github-token needs to be added to the repository with at least `read` access?

Reply:
From what I just tested - https://github.com/srv-rr-github-token doesn't have to have explicit access to the repo. Having an internal repo - created as a Splunk org - should be enough.